metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/LineageUtils.java

```diff
@@ -25,6 +38,10 @@ private LineageUtils() {
 
   }
 
+  public static Urn dataPlatformInstanceUrn(String platform, String instance) throws URISyntaxException {
+    return new Urn("urn:li:dataPlatformInstance:(" + new DataPlatformUrn(platform).toString() + "," + instance + ")");
+  }
+
   public static DataFlowUrn flowUrn(String master, String appName) {
     return new DataFlowUrn("spark", appName, master.replaceAll(":", "_").replaceAll("/", "_").replaceAll("[_]+", "_"));
```

Comment:
This can be done with a single replaceAll with Regex using a matcher for these three characters

Reply:
regex way seems more complicated for simple replace. this is more readable.

Reply:
It's already doing regex though and this does 3 full scans through the String and creates two intermediate Strings that get committed to the String pool as extra memory... I also disagree wholeheartedly that 

master.replaceAll("[:\\/] | \\[_\\]", "_")

is less readable than the current approach, but really don't let this block the PR. It was more of a nitpick in the first place 😄 hopefully the compiler just figures it out anyway.